contrib/mesos/pkg/scheduler/executorinfo/lru_cache.go

```diff
@@ -0,0 +1,95 @@
+/*
+Copyright 2015 The Kubernetes Authors All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package executorinfo
+
+import (
+	"container/list"
+	"errors"
+
+	"github.com/mesos/mesos-go/mesosproto"
+)
+
+// Cache is an LRU cache for executor info objects.
+// It is not safe for concurrent use.
+type Cache struct {
+	maxEntries int
+	ll         *list.List
+	cache      map[string]*list.Element // by hostname
```

Comment:
future todo: if this impacts scheduling throughput (because linked lists are slow) we could use a heap here instead (one that sorts on entry timestamp)